src/target_s3_json/snowflake.py

```diff
@@ -0,0 +1,163 @@
+        """Execute a Snowflake query using stored connection parameters."""
+        conn = snowflake.connector.connect(**self._connection_params)
+        
+        try:
+            cur = conn.cursor(snowflake.connector.DictCursor)
+            cur.execute(query)
+            results = cur.fetchall()
+            return results
+        finally:
+            cur.close()
+            conn.close()
+    
+    def _clean_identifier(self, value: str) -> str:
+        """Clean an identifier by removing hyphens and converting to uppercase."""
+        return value.replace('-', '').upper()
+    
+    def _create_stage_name(self) -> str:
+        """Create a Snowflake stage name in the format S3_STAGE"""
+        return "S3_STAGE"
+    
+    def _create_schema_name(self) -> str:
+        """Create a Snowflake schema name in the format T_ORGID_SOURCE."""
+        return f"T_{self._clean_identifier(self.path_components.org_id)}_{self._clean_identifier(self.path_components.source)}"
+    
+    def create_schema(self) -> None:
+        """Create a Snowflake schema if it doesn't exist."""
+        query = f"""
+        CREATE SCHEMA IF NOT EXISTS {self.schema_name};
+        """
+        self.execute_query(query)
+        LOGGER.info(f"Created or verified schema: {self.schema_name}")
+    
+    def create_s3_stage(self, s3_bucket: str) -> None:
+        """Create a Snowflake stage for S3 integration if it doesn't exist."""
+        # First create the schema if it doesn't exist
+        self.create_schema()
+        
+        query = f"""
+        CREATE STAGE IF NOT EXISTS {self.schema_name}.{self.stage_name}
+        STORAGE_INTEGRATION = s3_int
+        URL = 's3://{s3_bucket}/{self.path_components.org_id}/{self.path_components.source}'
+        DIRECTORY = (
+            ENABLE = TRUE, 
+            REFRESH_ON_CREATE = FALSE
+        )
+        FILE_FORMAT = (TYPE = JSON);
+        """
+        self.execute_query(query)
+        LOGGER.info(f"Created or verified stage: {self.schema_name}.{self.stage_name}")
+    
+    def refresh_directory(self) -> None:
+        """
+        Refresh a Snowflake directory for the stage.
+        For paths with repo_ids (e.g., GitHub), refreshes with that specific subpath/repo.
+        For datepaths with dates (e.g., Azure Tickets), refreshes the entire stage.
+        """
+        stage_ref = f"{self.schema_name}.{self.stage_name}"
+        
+        # Build the refresh query based on repo_id presence
+        if self.path_components.repo_id:
+            subpath = f"SUBPATH = '{self.path_components.repo_id}'"
+            LOGGER.info(f"Refreshing {stage_ref} with repo {self.path_components.repo_id}")
+        else:
+            subpath = ""
+            LOGGER.info(f"Refreshing entire {stage_ref}")
+            
+        query = f"ALTER STAGE {stage_ref} REFRESH {subpath};"
+        self.execute_query(query)
+
+def parse_path_template(path_template: str) -> PathComponents:
```

Comment:
No need to change as this parsing looks just fine. 

I used the python library [`parse`](https://pypi.org/project/parse/) to parse the paths before.  You craft a string template and then parse will pass back and object with the components that fit the scheme (https://github.com/minwareco/platform-sdk/blob/6e85a597821960a2dfc4d469b6e8499a09ace7a6/src/ingest/IngestJobConfigGenerator.ts#L542 for the filename and then https://github.com/minwareco/scheduled/blob/85e81ead9aaa614770b9f19507f5a43119c8485e/src/worker/s3tap.py#L98 for its usage).

Just thought it was worth mentioning, no need to change here though.  I found it useful because the part and stream are in the filename and not just splitting on directories.